### DISCUSSION TITLE
AutoMapper should user the Request.ODataFeature() extension to access features

### DISCUSSION THREAD

**Discussion by anhaehne** [AutoMapper/AutoMapper.Extensions.OData#179] [Ideas]
In the microsoft documention there are examples that show how to override some odata clauses on the server side. For example:
``` csharp
public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
{
    options.Request.ODataFeature().SelectExpandClause = new SelectExpandQueryOption(null, "Orders", options.Context,
        new ODataQueryOptionParser(
            model: options.Context.Model,
            targetEdmType: options.Context.NavigationSource.EntityType(),
            targetNavigationSource: options.Context.NavigationSource,
            queryOptions: new Dictionary<string, string> { { "$expand", "Orders" } },
            container: options.Context.RequestContainer)).SelectExpandClause;

    IQueryable results = options.ApplyTo(Customers.AsQueryable());

    return results as IQueryable<Customer>;
}
```
Source: [MSDN: SelectExpand query](https://learn.microsoft.com/en-us/odata/webapi-8/fundamentals/query-options?tabs=net60#selectexpand-query)

We can see that the `SelectExpandClause` is overriden. The problem is that automapper will use the `SelectExpandClause` from the `ODataQueryOptions<T>`. This will prevent the example from working with automapper.

My recommendation would be to change Automapper to access the clauses via the `IODataFeature` interface provided by the OData library.

**Comment by BlaiseD** [comment]
Consider submitting a PR with your recommended approach showing:

- All existing tests passing.
- The new tests which fail for the current approach and pass with the new approach.
